doc/coq.rst

```diff
@@ -0,0 +1,507 @@
+.. _coq:
+
+***
+Coq
+***
+
+
+Dune is able to build Coq projects and plugins with support for source file
+preprocessing, extraction, and documenation.
+
+
+.. contents:: Table of Contents
+    :depth: 3
+
+.. _coq-theory:
+
+coq.theory
+----------
+
+A *Coq theory* is a collection of ``.v`` files. Together with OCaml libraries
+linking to the Coq API, a Coq theory is termed a *Coq plugin*. Both may also be
+termed a *Coq Project*. To enable Coq support in a Dune project. The language
+version should be selected in the :ref:`dune-project` file. For example:
+
+.. code:: scheme
+
+    (using coq 0.3)
+
+This will allow for the ``coq.theory`` stanza to be included in the current
+project.
```

Comment:
Actually all the `(coq.* ...)` stanzas, which we have a few.

Reply:
Yes I noticed this. Do we want to list them one-by-one or just say `coq.*`?